docs/extending/extending_draftail.md

```diff
@@ -329,17 +325,203 @@ To fully complete the demo, we can add a bit of JavaScript to the front-end in o
 
 ```javascript
 document.querySelectorAll('[data-stock]').forEach((elt) => {
-    const link = document.createElement('a');
-    link.href = `https://finance.yahoo.com/quote/${elt.dataset.stock}`;
-    link.innerHTML = `${elt.innerHTML}<svg width="50" height="20" stroke-width="2" stroke="blue" fill="rgba(0, 0, 255, .2)"><path d="M4 14.19 L 4 14.19 L 13.2 14.21 L 22.4 13.77 L 31.59 13.99 L 40.8 13.46 L 50 11.68 L 59.19 11.35 L 68.39 10.68 L 77.6 7.11 L 86.8 7.85 L 96 4" fill="none"></path><path d="M4 14.19 L 4 14.19 L 13.2 14.21 L 22.4 13.77 L 31.59 13.99 L 40.8 13.46 L 50 11.68 L 59.19 11.35 L 68.39 10.68 L 77.6 7.11 L 86.8 7.85 L 96 4 V 20 L 4 20 Z" stroke="none"></path></svg>`;
+  const link = document.createElement('a');
+  link.href = `https://finance.yahoo.com/quote/${elt.dataset.stock}`;
+  link.innerHTML = `${elt.innerHTML}<svg width="50" height="20" stroke-width="2" stroke="blue" fill="rgba(0, 0, 255, .2)"><path d="M4 14.19 L 4 14.19 L 13.2 14.21 L 22.4 13.77 L 31.59 13.99 L 40.8 13.46 L 50 11.68 L 59.19 11.35 L 68.39 10.68 L 77.6 7.11 L 86.8 7.85 L 96 4" fill="none"></path><path d="M4 14.19 L 4 14.19 L 13.2 14.21 L 22.4 13.77 L 31.59 13.99 L 40.8 13.46 L 50 11.68 L 59.19 11.35 L 68.39 10.68 L 77.6 7.11 L 86.8 7.85 L 96 4 V 20 L 4 20 Z" stroke="none"></path></svg>`;
 
-    elt.innerHTML = '';
-    elt.appendChild(link);
+  elt.innerHTML = '';
+  elt.appendChild(link);
 });
 ```
 
 Custom block entities can also be created (have a look at the separate [Draftail documentation](https://www.draftail.org/docs/blocks)), but these are not detailed here since [StreamField](streamfield_topic) is the go-to way to create block-level rich text in Wagtail.
 
+## Other editor extensions
+
+Draftail has additional APIs for more arbitrary customisations:
+
+- **Controls** – To add arbitrary UI elements to editor toolbars.
+- **Decorators** – For arbitrary text decorations / highlighting.
+- **Plugins** – For direct access to all Draft.js APIs.
+
+### Custom toolbar controls
+
+To add an arbitrary new UI element to editor toolbars, Draftail comes with a [controls API](https://www.draftail.org/docs/arbitrary-controls). Controls can be arbitrary React components, which can get and set the editor state. Note controls update on _every keystroke_ in the editor – make sure they render fast!
+
+Here is an example with a simple sentence counter – first, registering the editor feature in a `wagtail_hooks.py`:
```

Comment:
I chose this example because it’s simple to understand the utility and simple to write. It doesn’t demonstrate setting the editor state, which would be nice but would instantly make the example 2-3x longer.